src/common/security/local/context.go

```diff
@@ -89,20 +90,24 @@ func (s *SecurityContext) HasReadPerm(projectIDOrName interface{}) bool {
 
 	roles := s.GetProjectRoles(projectIDOrName)
 
-	return len(roles) > 0
+	if len(roles) > 0 {
+		return true
+	}
+
+	//Get Group Roles
+	groupRoles := s.GetRolesByGroup(projectIDOrName)
```

Comment:
The result of GetProjectRoles() has already contained the group roles. No need to call GetRolesByGroup() again.
The same for HasWritePerm() and HasAllPerm()

Reply:
Fixed it

Reply:
@stonezdj So do we still need  `GetRolesByGroup`?

Reply:
This method is used in by GetProjectRoles(), still need it

Reply:
No need to call GetRolesByGroup() here.

Reply:
Removed it, issue maybe caused by code merge